### DISCUSSION TITLE
Diskussionen Backend #1

### DISCUSSION THREAD

**Discussion by Level8Broccoli** [Level8Broccoli/WebE#23] [Q&A]
- [x] Code-Kommentare sind z.T. Deutsch und z.T. Englisch. Wollen wir uns auf Englisch einigen?

- [x] Sagen diese Art von Kommentaren überhaupt etwas aus? (mehr als der Code auf der nächsten Zeile aussagt)

```js
  // Register new player endpoint
  socket.on("registerPlayer", (request: RegisterPlayerRequest) => {
```

- [ ] Hier bin ich mir nicht ganz sicher: Du wirfst ab und zu einen Error innerhalb eines Promises. Wäre für das nicht das `reject` Objekt?

```js
return new Promise((resolve, reject) => {
  // [Server] Validation (nicht leer, gültige UTF-8 Zeichen) -> Errorfeedback
  if (request.playerName.length === 0 || request.playerName === "") {
    throw new Error("Player name is empty.");
  }
```

- [ ] Ist das hier `request.playerName.length === 0` und `request.playerName === ""` nicht exakt dasselbe? Alternativer Vorschlag: `request.playerName.trim().length === 0`, dann fängt das auch Fälle wie `"      "` ab. Vielleicht lohnt es sich so oder so die `.trim()` Funktion auf alle User-Eingaben anzuwenden...

- [ ] Wollten wir nicht bei den Errors verschiedene Error-Codes verwenden? Wir müssen die Fehler dann auf dem GUI in zwei unterschiedliche Sprachen kommunizieren; das geht dann evtl. leichter mit einheitlichen Fehlercodes.

- [x] Wie besprochen, lohnt es wohl auf LUXON (https://github.com/moment/luxon/) umzusteigen (sowohl im Front- als auch Backend)

- [x] `ErrorCode` ist evtl. ein bisschen inkorrekt, da einer der Codes für *keinen Fehler* steht.

- [ ] Meinst du würden hier `Config` und `maxPlayerCount` auch reichen für die Bezeichnung?

```js
export type GameConfig = {
  maxPlayerCountForGame: number;
};
```

- [x] Solche Codestellen: `found !== undefined ? true : false` kann man abkürzen: `found !== undefined`. Ist exakt dasselbe.

- [x] Gibt es einen Grund, weshalb das Secret nicht auch einfach eine UUID(v4) ist? Ich hätte das aus Faulheit gleich so übernommen. Die Methode `_randomNumber()` scheint mir ein bisschen zu over-engineered - brauchen wir sonst noch wo Zufallszahlen dieser Art?

**Comment by ThierryGirod** [comment]
Englische Kommentare passt, da auch der Code sonst in Englisch gehalten ist.
Deutsche Kommentare waren lediglich da, um die Abläufe 1 zu 1 abzudecken.
Werden auf Englisch übersetzt

**Comment by ThierryGirod** [comment]
Luxon wird eingebaut.

**Comment by ThierryGirod** [comment]
Bezüglich Secret: Nein gibt keinen speziellen Grund :) Ich wollte einfach nicht, dass beide Id's identisch aussehen. 
Ändere es auf ebenfalls UUID v4 ab.
